### PR TITLE
Added firstDay option to datepicker control

### PR DESCRIPTION
**Pull request by arvislacis**
Added `firstDay` (first day of the week) option to datepicker control for Builder plugin. `firstDay` option itself was introduced in https://github.com/octobercms/october/pull/2940

### CODE REVIEW

File: classes/StandardControlsRegistry.php [hunk 1]
```diff
@@ -734,6 +734,22 @@ protected function registerDatepickerWidget()
                 'type' => 'string',
                 'ignoreIfEmpty' => true,
                 'sortOrder' => 85
+            ],
+            'firstDay' => [
+                'title' => Lang::get('winter.builder::lang.form.property_datepicker_first_day'),
+                'type' => 'dropdown',
+                'default' => 'datetime',
```
Comment (bennothommo) [inline classes/StandardControlsRegistry.php:741]: @arvislacis This needs to default to Sunday (`"0"`).

```suggestion
                'default' => '0',
```

File: classes/StandardControlsRegistry.php [hunk 2]
```diff
@@ -734,6 +734,22 @@ protected function registerDatepickerWidget()
...
+                'options' => [
+                    '0' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_sunday'),
+                    '1' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_monday'),
+                    '2' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_tuesday'),
+                    '3' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_wednesday'),
+                    '4' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_thursday'),
+                    '5' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_friday'),
+                    '6' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_saturday')
+                ],
+                'ignoreIfEmpty' => true,
```
Comment (bennothommo) [inline classes/StandardControlsRegistry.php:751]: @arvislacis Can you double-check that setting it to Sunday works? I might be mistaken, but a string `"0"` is still considered empty and might be stripped from the widget config.

File: classes/StandardControlsRegistry.php [hunk 3]
```diff
@@ -734,6 +734,22 @@ protected function registerDatepickerWidget()
...
+                'sortOrder' => 86
```
Comment (bennothommo) [inline classes/StandardControlsRegistry.php:752]:
```suggestion
                'sortOrder' => 86,
```

File: classes/StandardControlsRegistry.php [hunk 4]
```diff
@@ -734,6 +734,22 @@ protected function registerDatepickerWidget()
...
+                'type' => 'dropdown',
+                'default' => 'datetime',
+                'options' => [
+                    '0' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_sunday'),
+                    '1' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_monday'),
+                    '2' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_tuesday'),
+                    '3' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_wednesday'),
+                    '4' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_thursday'),
+                    '5' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_friday'),
+                    '6' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_saturday')
```
Comment (bennothommo) [inline classes/StandardControlsRegistry.php:749]:
```suggestion
                    '6' => Lang::get('winter.builder::lang.form.property_datepicker_first_day_saturday'),
```

**Review by bennothommo — CHANGES_REQUESTED** [review]

File: classes/StandardControlsRegistry.php (hunk 2 above)
Comment (arvislacis) [reply classes/StandardControlsRegistry.php:752]: @bennothommo I double-checked and "0" is not considered empty - if you set or leave it to Sunday (default) then the value `0` is written into `fields.yaml` file.

So I added another commit https://github.com/wintercms/wn-builder-plugin/pull/4/commits/04c0094ad640cf3f26dc1d631de5493e25bd0a80 which adds default option with empty string `""`.

Also sorry for `default' => 'datetime',` - I was looking to another option as an example and accidentally took default value from it.

Comment (arvislacis) [reply classes/StandardControlsRegistry.php:752]: In my opinion it's better to don't force Sunday and "0" value for all users as default option, especially if it's written into `fields.yaml` as I mentioned... Although it's very unlikely there also may be situation in the future where the system's default the first day of the week value will be changed or some global, per user configuration will be invented. So empty string `''` would be better choice here.

**Comment by arvislacis** [comment]
@bennothommo Anything else needed from my side?

**Comment by bennothommo** [comment]
@arvislacis Nope, just needs to be tested.

**Comment by arvislacis** [comment]
@bennothommo Maybe I can help with testing somehow or (so this can be merged and closed)? Overall I don't see any major points where and how this pull request could break things or don't work, especially after the last commit which sets default value to empty string (so no default/existing configurations are changed either). For one of my personal projects this worked without any issues. :)

**Comment by bennothommo** [comment]
@arvislacis Looks good to me. Merging.

**Pull request merged**